src/main/java/de/orchound/gameoflife/model/Game.java

```diff
@@ -66,11 +69,29 @@ public void toggleCell(Vector2ic cell) {
 		if (!cellInBoardRange(cell))
 			return;
 
-		if (board.getCellStatus(cell.y(), cell.x()))
+		if (board.getCellStatus(cell.y(), cell.x())) {
 			board.killCell(cell.y(), cell.x());
-		else
+			paintMode = false;
+		}
+		else {
 			board.resurrectCell(cell.y(), cell.x());
+			paintMode = true;
+		}
+		notifyBoardDataObservers();
 
+
+	}
+
+	public void setCell(Vector2ic cell) {
+		if (paintMode) {
+			if (!board.getCellStatus(cell.y(), cell.x())) {
+				board.resurrectCell(cell.y(), cell.x());
+			}
+		} else {
+			if (board.getCellStatus(cell.y(), cell.x())) {
+				board.killCell(cell.y(), cell.x());
+			}
+		}
 		notifyBoardDataObservers();
```

Comment:
Notifying the observers here is somewhat costly and should only be done if the the board state actually changed.
This is currently not the case. The function might end up not changing the state at all and still communicates an update to the observers.